llvm/lib/CodeGen/AsmPrinter/OxCamlGCPrinter.cpp

```diff
@@ -173,10 +187,16 @@ bool OxCamlGCMetadataPrinter::emitStackMaps(Module &M, StackMaps &SM, AsmPrinter
 
     // frame_data
     uint64_t FrameSize = CSI.CSFunctionInfo.StaticStackSize;
-    if (CSI.ID != StatepointDirectives::DefaultStatepointID)
-      FrameSize += CSI.ID; // Stack offset from OxCaml
     FrameSize += PtrSize; // Return address
 
+    if (CSI.ID != StatepointDirectives::DefaultStatepointID) {
```

Comment:
This condition relies on the current value of `DefaultStatepointID` being `0xABCDEF00` not clashing with any of the legalling encoded values: if any of the top 16 bit are non-zero, then the bottom bit is set. This seems fine. Also looks like on the ocaml side we always pass something for ID, and the "default" is 0 if not adjustment is needed for the dynamic frames... Maybe just add a brief comment here about this frame size adjustment.